src/ray/raylet/worker_pool.cc

```diff
@@ -51,13 +48,6 @@ WorkerPool::WorkerPool(int num_workers, int maximum_startup_concurrency,
     : maximum_startup_concurrency_(maximum_startup_concurrency),
       gcs_client_(std::move(gcs_client)) {
   RAY_CHECK(maximum_startup_concurrency > 0);
-#ifdef _WIN32
-  // TODO(mehrdadn): Is there an equivalent of this we need for Windows?
-#else
-  // Ignore SIGCHLD signals. If we don't do this, then worker processes will
-  // become zombies instead of dying gracefully.
-  signal(SIGCHLD, SIG_IGN);
```

Comment:
How is this handled with boost::process ?

Reply:
I believe it just makes sure to call `waitpid` so there isn't an issue. I'm pretty sure this manipulating signal handlers is the wrong way to try to handle it. :-)